internal/component/loki/process/stages/sampling.go

```diff
@@ -66,14 +66,15 @@ type samplingStage struct {
 
 func (m *samplingStage) Run(in chan Entry) chan Entry {
 	out := make(chan Entry)
+	reason := *m.cfg.DropReason
```

Comment:
I'm not sure this is sufficient. I think `m.isSampled` may give the wrong result too. I think I need to make a local for all of m.